File: Library/Classes/Core/Categories/UIKit/UITableViewCell+CSExtension.m
//
//  Created by Rene Dohan on 5/7/12.
//
#import "CSLang.h"
#import "UIView+CSExtension.h"
#import "UITableViewCell+CSExtension.h"


@implementation UITableViewCell (CSExtension)


- (void)setBackgroundViewColor:(UIColor *)color {
    self.backgroundView = [UIView withColor:color];
}

- (void)setSelectedBackgroundColor:(UIColor *)color {
    self.selectedBackgroundView = [UIView withColor:color];;
}

- (UIView *)cellView {
	return self.contentView.content;
}

@end

File: Renetik/Renetik/Classes/Core/Extensions/UIKit/UITableViewCell+CSExtension.swift
//
// Created by Rene Dohan on 1/22/20.
//

import UIKit

public extension UITableViewCell {
    @discardableResult
    override open func construct() -> Self {
        super.construct()
        selectedBackgroundView = UIView.construct(color: UIColor.clear)
        contentView.matchParent()
        return self
    }

    func setBackgroundViewColor(_ color: UIColor) {
        backgroundView = UIView.construct(color: color)
    }

    func setSelectedBackgroundColor(_ color: UIColor) {
        selectedBackgroundView = UIView.construct(color: color)
    }

    var cellView: UIView? {
        contentView.content
    }
}
